### PR TITLE
Caramel Food

### PR DESCRIPTION
**Pull request by bathosbathtime**
Caramel foods again.  Caramel burger, caramel cookies, caramel cake, caramel pie, caramel apples, and caramel recipe.  Didn't touch flans this time.  Caramel condiment holder sprite "improved."  Will update the wiki once it is merged to avoid confusion.
Compiled and tested locally.
caramelfoods

:cl:
 * rscadd: caramel burgers, caramel cookies, caramel cake, caramel pie, caramel apples, and caramel itself can now be made in the microwave

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -1112,6 +1124,19 @@
...
+/obj/item/weapon/reagent_containers/food/snacks/pie/caramelpie
+	name = "caramel pie"
+	desc = "Sweet pie made with caramel"
+	icon_state = "pie"
+	trash = /obj/item/trash/pietin
+	food_flags = FOOD_SWEET
+
+/obj/item/weapon/reagent_containers/food/snacks/pie/caramelpie/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 4)
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: put `reagents.clear_reagents()` right before this

Comment (bathosbathtime) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: Added, testing now

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -2785,6 +2822,29 @@
...
+	storage_slots = 3
+	w_class = W_CLASS_MEDIUM
+	food_flags = FOOD_SWEET | FOOD_ANIMAL | FOOD_LACTOSE
+
+/obj/item/weapon/reagent_containers/food/snacks/sliceable/caramelcake/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 15)
+	reagents.add_reagent(CARAMEL, 5)
+
+/obj/item/weapon/reagent_containers/food/snacks/chocolatecakeslice
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: you forgot to change this to caramelcakeslice

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1137]: put `reagents.clear_reagents()` right before this

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -2785,6 +2822,29 @@
 	bitesize = 2
 	food_flags = FOOD_SWEET | FOOD_ANIMAL | FOOD_LACTOSE
 
+/obj/item/weapon/reagent_containers/food/snacks/sliceable/caramelcake
+	name = "caramel cake"
+	desc = "A cake with added caramel"
+	icon_state = "caramelcake"
+	slice_path = /obj/item/weapon/reagent_containers/food/snacks/chocolatecakeslice
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: you forgot to change this to caramelcakeslice

**Review by Intigracy — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (bathosbathtime) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: wow that's really bad

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 3 above)
Comment (bathosbathtime) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: should be fixed now

**Review by Intigracy — APPROVED** [review]

**Comment by SonixApache** [comment]
dumb changelog

**Pull request merged**